src/jit/emitxarch.cpp

```diff
@@ -6599,15 +6610,28 @@ void emitter::emitIns_R_S(instruction ins, emitAttr attr, regNumber ireg, int va
     emitAttr size = EA_SIZE(attr);
     noway_assert(emitVerifyEncodable(ins, size, ireg));
 
+    UNATIVE_OFFSET sz;
     instrDesc*     id  = emitNewInstr(attr);
-    UNATIVE_OFFSET sz  = emitInsSizeSV(id, insCodeRM(ins), varx, offs);
     insFormat      fmt = emitInsModeFormat(ins, IF_RRD_SRD);
 
+    id->idIns(ins);
+    id->idInsFmt(fmt);
+    id->idReg1(ireg);
+    id->idAddr()->iiaLclVar.initLclVarAddr(varx, offs);
+
+    sz = emitInsSizeSV(id, insCodeRM(ins), varx, offs);
+
     // Most 16-bit operand instructions need a prefix
     if (size == EA_2BYTE && ins != INS_movsx && ins != INS_movzx)
     {
         sz++;
     }
+    
+    if (Is4ByteSSEInstruction(ins))
```

Comment:
This is the actual targeted fix.

I'm going to see if I can find any other places that need it as well, but I also want to see if I can just get a proper fix of centralizing this check done.

Reply:
I ended up having to centralize various checks to ensure that we aren't underestimating and that we also don't egregiously overestimate.